src/odbc/odbc.c

```diff
@@ -4785,11 +4785,27 @@ ODBC_FUNC(SQLColumns, (P(SQLHSTMT,hstmt), PCHARIN(CatalogName,SQLSMALLINT),	/* o
 	WIDE))
 {
 	int retcode;
+	DSTR catalog_name = DSTR_INITIALIZER;
+	SQLSMALLINT catalog_size;
+	const char *proc = NULL;
 
 	ODBC_ENTER_HSTMT;
 
+	if (cbCatalogName == SQL_NTS) {
+		if (!odbc_dstr_copy(stmt->dbc, &catalog_name, cbCatalogName, szCatalogName))
+			goto memory_error;
+		catalog_size = tds_dstr_len(&catalog_name);
+	} else {
+		catalog_size = cbCatalogName;
+	}
+
+	proc = "sp_columns";
+	if (catalog_size > 0) {
+		proc = "..sp_columns";
+	}
+
 	retcode =
-		odbc_stat_execute(stmt _wide, "sp_columns", TDS_IS_MSSQL(stmt->dbc->tds_socket) ? 5 : 4,
+		odbc_stat_execute(stmt _wide, proc, TDS_IS_MSSQL(stmt->dbc->tds_socket) ? 5 : 4,
```

Comment:
I have the feeling that just replacing `"sp_columns"` with `"..sp_columns"` here will solve the problem. Please send the test case, I can integrate in some way and use to test all various server types and versions.

Reply:
Added to the PR; For the test, I modified the only one where I saw `SQLColumns` being tested - `stats` - it's pretty ugly, sorry.

Thanks for considering it.